VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsThreadFragment.java

```diff
@@ -219,12 +208,14 @@ public void onEventMainThread(DiscussionThreadPostedEvent event) {
                 }
             }
             discussionPostsAdapter.insert(event.getDiscussionThread(), i);
+            // In case this is the first addition, we need to hide the no-item-view
+            ((TaskProcessCallback) getActivity()).onMessage(MessageType.EMPTY, "");
         }
     }
 
     @Override
     public void loadNextPage(@NonNull final InfiniteScrollUtils.PageLoadCallback<DiscussionThread> callback) {
-        if (isFollowingTopics()) {
+        if (discussionTopic.isFollowingType()) {
```

Comment:
This check can be moved down to `GetThreadListTask` if we pass `DiscussionTopic` instead of a list of thread IDs. We could merge `GetThreadListTask` and `GetFollowingThreadListTask`. That would also allow us to remove a lot of duplicate logic in this class.

This would incidentally fix the fact that paging is currently broken for the "following" topic (nextPage is not incremented) but also prevent future issues from these two tasks not being treated equally.

I would also move the `getAllTopicIds` into `GetThreadListTask`  at that point since that's the only place it's used/needed.